brian2/core/variables.py

```diff
@@ -274,6 +275,9 @@ def __repr__(self):
                                   constant=repr(self.constant),
                                   read_only=repr(self.read_only))
 
+    _state_tuple = property(lambda self: (self.dim, self.dtype, self.scalar,
+                                          self.constant, self.read_only,
+                                          self.dynamic, self.name))
```

Comment:
Similar comment to above, is there any way we can make this more safe against our own future changes by automating the definition of ``_state_tuple``?